src/dataflow/core/lispress.py

```diff
@@ -685,3 +671,36 @@ def lispress_to_type_name(e: Lispress) -> TypeName:
         return TypeName(constructor, tuple([lispress_to_type_name(a) for a in args]))
     else:
         raise ValueError(f"unexpected lispress {e}")
+
+
+def _canonicalize_program(program: Program) -> Program:
```

Comment:
since `let` bindings are created in Program order, this has the effect of canonicalizing `let` bindings.